sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -275,6 +289,10 @@ class AstBuilder extends DataTypeAstBuilder
         if Option(c.beginLabel()).isDefined &&
           c.beginLabel().multipartIdentifier().getText.toLowerCase(Locale.ROOT).equals(label)
         => true
+      case c: RepeatStatementContext
+        if Option(c.beginLabel()).isDefined &&
+          c.beginLabel().multipartIdentifier().getText.toLowerCase(Locale.ROOT).equals(label)
+      => true
```

Comment:
Can we merge this with the above case for `WhileStatementContext`? You have multiple options for this, using `|` operator or `Either` type.

Reply:
This wouldn't be trivial, as there is no common supertype between `RepeatStatementContext` and `WhileStatementContext` with the `beginLabel()` method defined. Maybe we could have a `labeledStatement` grammar rule, or something similar, to abstract some of this label logic from all statements with labels (ifElse, while, repeat..)